src/frame/index.js

```diff
@@ -0,0 +1,19 @@
+import fledgeAPI from './api.js';
+
+const VERSION = 1;
+const VERSION_KEY = 'fledge.polyfill';
+
+export default async function main (debug = false) {
+	// check whenever the document is being framed by a site which you don’t expect it to be framed by
+	const [ parentOrigin ] = window.location.ancestorOrigins;
+	if (parentOrigin === undefined) {
+		throw new Error("Frame can't run as a top-level document");
+	}
+
+	// connect to the storage iframe and send a message
+	const { port1: receiver, port2: sender } = new MessageChannel();
+	receiver.onmessage = fledgeAPI;
+	window.parent.postMessage({
+		[VERSION_KEY]: VERSION,
+	}, parentOrigin, [ sender ]);
```

Comment:
this is a pretty clean API, it handles request/response automatically? You can `send message 1` `send message2` `receive message 2` `receive message 1` and it figures out responses to the right requests?

Reply:
yeah, i haven't figured out a way to test that as I'm still wrapping my head around it, but that's the general concept according to this article: https://javascript.plainenglish.io/message-channels-f94de3488b1a

Reply:
That article didn't mention anything about sequencing or request/response. It seems like message channels just provide a clean interface for sending and receiving one way messages. Much like email, the message itself needs to include information that indicates its a reply to a previous message or a brand new one. I think it might still be prudent to use a message ID as I had in the [prop cohort prototype.](https://github.com/MagniteEngineering/ProprietaryCohorts/blob/protoype/src/browser/proprietary-cohorts.js#L70)

Reply:
Can we maybe backlog this? If and when it becomes a problem, then we can address it, but I'd rather not restructure my code for something that hasn't been tested to be an issue at this moment.

Reply:
Created Issue #65